crates/red_knot/src/python_version.rs

```diff
@@ -0,0 +1,68 @@
+/// Enumeration of all supported Python versions
+///
+/// TODO: unify with the `PythonVersion` enum in the linter/formatter crates?
+#[derive(Copy, Clone, Hash, Debug, PartialEq, Eq, PartialOrd, Ord, Default, clap::ValueEnum)]
+pub enum PythonVersion {
+    #[value(name = "3.7")]
+    Py37,
+    #[value(name = "3.8")]
+    Py38,
+    #[default]
+    #[value(name = "3.9")]
+    Py39,
+    #[value(name = "3.10")]
+    Py310,
+    #[value(name = "3.11")]
+    Py311,
+    #[value(name = "3.12")]
+    Py312,
+    #[value(name = "3.13")]
+    Py313,
+}
+
+impl PythonVersion {
+    const fn as_str(self) -> &'static str {
+        match self {
+            Self::Py37 => "37",
+            Self::Py38 => "38",
+            Self::Py39 => "39",
+            Self::Py310 => "310",
+            Self::Py311 => "311",
+            Self::Py312 => "312",
+            Self::Py313 => "313",
```

Comment:
should these all now be dotted?

```suggestion
            Self::Py37 => "3.7",
            Self::Py38 => "3.8",
            Self::Py39 => "3.9",
            Self::Py310 => "3.10",
            Self::Py311 => "3.11",
            Self::Py312 => "3.12",
            Self::Py313 => "3.13",
```